libs/ledger-live-common/src/network.ts

```diff
@@ -1,17 +1,65 @@
-import invariant from "invariant";
-import axios, { AxiosPromise } from "axios";
-import type { AxiosError, AxiosRequestConfig, Method } from "axios";
+import { LedgerAPI4xx, LedgerAPI5xx, NetworkDown } from "@ledgerhq/errors";
 import { log } from "@ledgerhq/logs";
-import { NetworkDown, LedgerAPI5xx, LedgerAPI4xx } from "@ledgerhq/errors";
+import type { AxiosError, AxiosRequestConfig, Method } from "axios";
+import axios, { AxiosPromise, AxiosResponse } from "axios";
+import invariant from "invariant";
+import { changes, getEnv } from "./env";
 import { retry } from "./promise";
-import { getEnv, changes } from "./env";
+
+type Metadata = { startTime: number };
+type ExtendedXHRConfig = AxiosRequestConfig & { metadata?: Metadata };
+
+export const requestInterceptor = (
+  request: AxiosRequestConfig
+): ExtendedXHRConfig => {
+  if (!getEnv("ENABLE_NETWORK_LOGS")) {
+    return request;
+  }
+
+  const { baseURL, url, method = "", data } = request;
+  log("network", `${method} ${baseURL || ""}${url}`, { data });
+
+  const req: ExtendedXHRConfig = request;
+
+  req.metadata = {
+    startTime: Date.now(),
+  };
+
+  return req;
+};
+
+type Response = {
+  config: ExtendedXHRConfig;
+} & AxiosResponse<any>;
+
+export const responseInterceptor = (response: Response): Response => {
+  if (!getEnv("ENABLE_NETWORK_LOGS")) {
+    return response;
+  }
+
+  const { baseURL, url, method = "", metadata } = response.config;
+  const { startTime = 0 } = metadata || {};
+
+  log(
+    "network-success",
+    `${response.status} ${method} ${baseURL || ""}${url} (${(
+      Date.now() - startTime
+    ).toFixed(0)}ms)`,
+    { data: response.data }
+  );
+
+  return response;
+};
 
 export const errorInterceptor = (error: AxiosError<any>): AxiosError<any> => {
-  const config = error?.response?.config || null;
+  const config = error?.response?.config as ExtendedXHRConfig | null;
```

Comment:
not sure but if `|` is same as `||` ? I though `|` was more for typing

Reply:
This doesn't look equivalent :grimacing: 

error?.response?.config returns undefined if config is not defined

Which means config before would be equal to null, now it's equal to undefined

![image](https://github.com/LedgerHQ/ledger-live/assets/19901996/da8111c7-e1a8-432e-b2a1-a70120fdd376)

Maybe the change has no impact though, but it needs to be checked

Reply:
It's not the same indeed. Here `|` is used because we are in the context of TS types (we are casting `error?.response?.config` as a `ExtendedXHRConfig | null` type).

Here, the type of `config` is `ExtendedXHRConfig | null`.

An alternative, although not identical, would be:
```ts
const config = (error?.response?.config as ExtendedXHRConfig) || null;
```
In this case, `error?.response?.config` is casted as a `ExtendedXHRConfig` type. The `|| null` is superflus since `config` will always have type `ExtendedXHRConfig` (because TS casting juste changes the type of a value, without checking if it works or not. This is not an operation).

Reply:
I fixed the typing here 59e26dabfdc2c003d256771c344a9437842bfac6 to get rid of this casting and improve clarity :+1: